src/spacy/app.clj

```diff
@@ -90,14 +81,30 @@
       {:consumes "application/x-www-form-urlencoded"
        :parameters {:form {:title String :description String}}
        :response (fn [ctx]
-                   (let [event-id (get-in ctx [:parameters :path :event-id])
+                   (let [current-user "joy"
+                         event-id (get-in ctx [:parameters :path :event-id])
                          params  (get-in ctx [:parameters :form])
-                         session (:session data)
+                         session (data/fetch data event-id)
+                         new-state (domain/suggest-session session current-user params)
                          channel (:channel events)]
-                     (let [old-state @session
-                           new-state (swap! session update-state params)]
-                       (publish-events! channel old-state new-state)
-                       (yada-redirect ctx (bidi/path-for routes ::event :event-id event-id)))))}}})))
+                     (data/persist! data event-id new-state)
+                     (yada-redirect ctx (bidi/path-for routes ::event :event-id event-id))))}}})))
+
+(defmulti ^:private interpret-fact ::domain/fact)
+
+(defmethod interpret-fact :default
+  [fact]
+  (log/warn ::unknown-fact fact))
+
+(defmethod interpret-fact ::domain/session-suggested
+  [fact]
+  (let [{::domain/keys [session sponsor]} fact
+        {::domain/keys [title description id]} session]
+    {::fact :session-suggested,
+     ::session {:id id
+                :title title
+                :description description
+                :sponsor sponsor}}))
```

Comment:
I think I actually unintentionally changed the structure of the session while implementing the frontend. I will maybe try to change the frontend later so that it more exactly matches the data structure of our domain objects.

Reply:
I'm also happy to revisit the spec of events. I'm not entirely sure if sessions should be nested the way they are now:

```clojure
{::sponsor "joy"
 ::session {::title "Responsive and Accessible"
            ::description "Responsible!"
            ::id (random-uuid)}}
```

Maybe we should flatten it.

Reply:
Hmm... I'm looking at the definition of the facts in `domain.spacy` at [L59](https://github.com/innoq/spacy/blob/main/src/spacy/domain.clj#L59) and [L67](https://github.com/innoq/spacy/blob/main/src/spacy/domain.clj#L67). I think we should decide for things like `::sponsor`, `::time`, and `::room` --> do they belong to the fact? Or to the session entity? If they belong to the fact, then our data structures make sense here. But if `::sponsor` (for instance) belongs to the session, it might make sense to pull it into the `::session` map.

Right now, I'm thinking that we could make a case for `::sponsor` being either a member of the fact or of the session, but I think `::time` and `::room` might belong more to the fact (as they are now), since a session does not always have a `::time` and `::room`